### PR TITLE
fix(dropdown): trigger keydown handler should only work for trigger element

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:**  #4066 #3880

## Summary

fix(dropdown): trigger keydown handler should only work for trigger element.

### CODE REVIEW

**Comment by driskull** [comment]
@jcfranco can you review these changes? What I'm doing seems to work, just wondering why the previous code was there.

File: src/components/dropdown/dropdown.tsx [hunk 1]
```diff
@@ -438,28 +438,27 @@ export class Dropdown implements InteractiveComponent {
...
-          break;
-      }
+
+    if (this.active && (key === "Escape" || (e.shiftKey && key === "Tab"))) {
+      this.closeCalciteDropdown();
+      return;
+    }
+
+    switch (key) {
+      case " ":
```
Comment (jcfranco) [inline src/components/dropdown/dropdown.tsx:455]: Sidebar: this could use the `isActionKey` util if we switch (ba dum tss 🥁 ) to an `if`/`else` statement.

**Review by jcfranco — APPROVED** [review]
LGTM! 

We'll assume existing tests will cover this since writing a test for all acceptable triggers would be a pain to maintain. 💥👊💥

I had one refactor suggestion, but that can be tackled separately.

**Comment by jcfranco** [comment]
cc @Elijbet

**Pull request merged**